index.bs

```diff
@@ -346,20 +346,99 @@ interface mixin DatagramTransport {
 
 <pre class="idl">
-interface mixin WebTransport {
+[Exposed=(Window,Worker)]
+interface WebTransport {
+  constructor(USVString url, optional WebTransportOptions options = {});
+
+  Promise&lt;WebTransportStats&gt; getStats();
   readonly attribute WebTransportState state;
   readonly attribute Promise&lt;WebTransportCloseInfo&gt; closed;
   void close(optional WebTransportCloseInfo closeInfo = {});
   attribute EventHandler onstatechange;
 };
+
+WebTransport includes UnidirectionalStreamsTransport;
+WebTransport includes BidirectionalStreamsTransport;
+WebTransport includes DatagramTransport;
 </pre>
 
+## Constructor ##  {#webtransport-constructor}
+
+When the {{WebTransport/constructor()}} constructor is invoked, the user
+agent MUST run the following steps:
+1. Let |parsedURL| be the [=URL record=] resulting from [=URL parser|parsing=]
+   of {{WebTransport/constructor(url, options)/url}}.
+1. If |parsedURL| is a failure, [=throw=] a {{SyntaxError}} exception.
+1. If |parsedURL| [=scheme=] is not `quic-transport` or `https`, [=throw=]
+   a {{SyntaxError}} exception.
+1. If |parsedURL| [=fragment=] is not null, [=throw=] a {{SyntaxError}} exception.
+1. Let |transport| be a newly constructed {{WebTransport}} object.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[OutgoingStreams]]</dfn> internal slot
+   representing a sequence of {{OutgoingStream}} objects, initialized to empty.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[ReceivedStreams]]</dfn> internal slot
+   representing a {{ReadableStream}} of {{IncomingStream}} objects, initialized
+   to empty.
+1. Let |transport| have a
+  <dfn attribute for="WebTransport">\[[ReceivedBidirectionalStreams]]</dfn>
+  internal slot representing a {{ReadableStream}} of {{IncomingStream}}
+  objects, initialized to empty.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[WebTransportState]]</dfn> internal
+   slot, initialized to `"connecting"`.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[SentDatagrams]]</dfn> internal slot
+   representing a {{WritableStream}} of {{Uint8Array}}s, initialized to empty.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[ReceivedDatagrams]]</dfn> internal
+   slot representing a {{ReadableStream}} of {{Uint8Array}}s, initialized to
+   empty.
+1. If the scheme of |parsedURL| is `quic-transport`, [=in parallel=],
+   [=initialize WebTransport over QUIC=].
+1. If the scheme of |parsedURL| is `https`, [=in parallel=],
+   [=initialize WebTransport over HTTP=].
+1. Run these steps in parallel:
+1. Return |transport|.
+
+<div algorithm="initialize WebTransport over QUIC">
+To <dfn>initialize WebTransport over QUIC</dfn> for a given |transport| and
+|parsedURL|, do the following:
+
+1. Let |clientOrigin| be |transport|'s [=relevant settings object=]'s
+   [=origin=],
+   [serialized](https://html.spec.whatwg.org/multipage/origin.html#ascii-serialisation-of-an-origin).
```

Comment:
Note for future updates: find a way to auto-linkify this.

Reply:
Done.